### ISSUE TITLE
1.4.2 LaTeX crashes if code-block inside warning directive.

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#2640]
Following 

```
Hello
=====


.. warning::

   we should be allowed to do a code-block inside a warning

   .. code-block:: python

      1+1
```

causes a crash. Sorry.

**Comment by jfbu** [comment]
Really tough latex debugging.

**Comment by jnothman** [comment]
> Really tough latex debugging.

I don't doubt it. Thanks very much, though.

**Comment by jfbu** [comment]
Merged into stable branch at 5ba3293. In the future, a better way should be used by sphinx.sty for its fancyvrb extended Verbatim. Rather than preserving original in OriginalVerbatim and patch the Verbatim (with the hidden constraint that no `\begin{foo}` could be used --  fixed by this PR), it should use new name like SphinxVerbatim and leave Verbatim to original; by the way fancyvrb provides facilities to do that.

**Comment by jfbu** [comment]
@jnothman it was tough for me because the thrown error looked like some other issues with nested lists, but in the end it was due to using \OriginalVerbatim/\endOriginalVerbatim syntax which does not set the "current environment" , and the Verbatim part was fooled by surrounding \begin{minipage}/\end{minipage} (*)  failed to identify its ending and executed some code where it should never have been. The first thrown error was then TeX typically misleading. Thanks for reporting !

(*) I do not know a simple way to nest breakable framed structures in LaTeX and framed.sty is no worse than more recent packages (e.g. http://tex.stackexchange.com/questions/179952/should-tcolorbox-enforce-inner-boxes-to-be-unbreakable), and Sphinx uses minipage if the framed block is inside another framed (breakable) block (warnings, topic boxes, and of course a code-block will not contain any sub box as it is "verbatim") as a necessity.
